File: back/src/models/ServantEvaluation.js
import mongoose from 'mongoose';

const servantEvaluationSchema = new mongoose.Schema(
  {
    templateId: { type: String, required: true, index: true },
    servantUsername: { type: String, required: true, index: true },
    weekDate: { type: String, required: true, index: true },
    value: { type: mongoose.Schema.Types.Mixed, default: false },
    scannedAt: { type: String, default: () => new Date().toISOString() },
    serviceYear: { type: String, default: '2026', index: true },
  },
  {
    timestamps: true,
    strict: false,
    collection: 'servantEvaluations',
  }
);

servantEvaluationSchema.index({ servantUsername: 1, templateId: 1, weekDate: 1 });

const ServantEvaluation = mongoose.models.ServantEvaluation || mongoose.model('ServantEvaluation', servantEvaluationSchema);
export default ServantEvaluation;

File: back/Models/ServantEvaluation.cs
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace Eftekad.Backend.Models
{
    [Table("servantEvaluations")]
    public class ServantEvaluation
    {
        [Key]
        [Column("id")]
        public string Id { get; set; } = string.Empty;

        [Column("templateId")]
        public string TemplateId { get; set; } = string.Empty;

        [Column("servantUsername")]
        public string ServantUsername { get; set; } = string.Empty;

        [Column("weekDate")]
        public string WeekDate { get; set; } = string.Empty;

        [Column("valueJson")]
        public string ValueJson { get; set; } = "false"; // Stored as JSON string (mixed value: bool, number, etc.)

        [Column("scannedAt")]
        public string ScannedAt { get; set; } = DateTime.UtcNow.ToString("o");

        [Column("serviceYear")]
        public string ServiceYear { get; set; } = "2026";

        [Column("createdAt")]
        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

        [Column("updatedAt")]
        public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;
    }
}
